libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/restarea/api/MapboxRestAreaApi.kt

```diff
@@ -0,0 +1,191 @@
+     */
+    fun generateRestAreaGuideMap(
+        instructions: BannerInstructions,
+        consumer: MapboxNavigationConsumer<Expected<RestAreaGuideMapError, RestAreaGuideMapValue>>
+    ) {
+        val action = RestAreaAction.CheckRestAreaMapAvailability(instructions)
+        when (val result = RestAreaProcessor.process(action)) {
+            is RestAreaResult.RestAreaMapAvailable -> {
+                requestGuideMap(result, consumer)
+            }
+            is RestAreaResult.RestAreaMapUnavailable -> {
+                onError(
+                    message = "No service/parking area guide map available.",
+                    consumer = consumer
+                )
+            }
+            else -> {
+                onError(
+                    message = "Inappropriate $result emitted for $action.",
+                    consumer = consumer
+                )
+            }
+        }
+    }
+
+    /**
+     * The method takes in [RouteProgress] and generates rest area guide map based on the presence
+     * of [RestStop.guideMapUri].
+     * @param routeProgress object representing [RouteProgress]
+     * @param consumer informs about the state of sapa guide map.
+     */
+    fun generateUpcomingRestAreaGuideMap(
+        routeProgress: RouteProgress,
+        consumer: MapboxNavigationConsumer<Expected<RestAreaGuideMapError, RestAreaGuideMapValue>>
+    ) {
+        val action = RestAreaAction.CheckUpcomingRestStop(routeProgress)
+        when (val result = RestAreaProcessor.process(action)) {
+            is RestAreaResult.RestAreaMapAvailable -> {
+                requestGuideMap(result, consumer)
+            }
+            is RestAreaResult.RestAreaMapUnavailable -> {
+                onError(
+                    message = "No service/parking area guide map available.",
+                    consumer = consumer
+                )
+            }
+            else -> {
+                onError(
+                    message = "Inappropriate $result emitted for $action.",
+                    consumer = consumer
+                )
+            }
+        }
+    }
+
+    /**
+     * Invoke the method to cancel all ongoing requests to generate a sapa guide map.
+     */
+    fun cancelAll() {
+        mainJobController.job.cancelChildren()
+    }
+
+    private fun requestGuideMap(
+        result: RestAreaResult.RestAreaMapAvailable,
+        consumer: MapboxNavigationConsumer<Expected<RestAreaGuideMapError, RestAreaGuideMapValue>>
+    ) {
+        val url = Uri.parse(result.sapaMapUrl).buildUpon().apply {
+            appendQueryParameter(KEY, accessToken)
+        }.build().toString()
+        val requestAction = RestAreaAction.PrepareRestAreaMapRequest(url)
+        val signboardRequest = RestAreaProcessor.process(requestAction)
+        val loadRequest = (signboardRequest as RestAreaResult.RestAreaMapRequest).request
+        mainJobController.scope.launch {
+            val loadResult = resourceLoader.load(loadRequest)
```

Comment:
Is this supposed to be called from the main thread? I don't see any proxying to the worker thread on our side, is it done on the NN side?